script.js

```diff
@@ -1,4 +1,160 @@
+    (userChoose == "scissors" && aiChoose == "paper") ||
+    (userChoose == "paper" && aiChoose == "stone")
+  ) {
+    totalPlays = totalPlays + 1;
+    userScore = userScore + 1;
+    console.log(userScore);
+    myOutputValue =
+      "You have won! <br><br> You have won " +
+      userScore +
+      " out of " +
+      totalPlays +
+      " times";
+  }
+  if (userChoose == "score check") {
+    myOutputValue =
+      " You have won " + userScore + " out of " + totalPlays + " times";
+  }
+  return myOutputValue;
+};
+
+/// Reverse SPS game mode ///
+var reverseSPS = function (userChoose) {
+  var myOutputValue =
+    "You are now playing Reverse Scissors Paper Stone. <br><br>The rules have changed! <br><br> Please input either 'Scissors', 'Paper' or 'Stone'. ";
+  var aiChoose = comPlays();
+  console.log("Com chooses " + aiChoose);
+  console.log("user chooses " + userChoose);
+  if (userChoose == aiChoose) {
+    totalPlays = totalPlays + 1;
+    myOutputValue = "It's a draw! Try again ";
+  }
+  if (
+    (userChoose == "stone" && aiChoose == "scissors") || //Computer wins User
+    (userChoose == "scissors" && aiChoose == "paper") ||
+    (userChoose == "paper" && aiChoose == "stone")
+  ) {
+    myOutputValue = "Aww. You have lost";
+    totalPlays = totalPlays + 1;
+    comScore = comScore + 1;
+    console.log(comScore);
+  }
+  if (
+    (userChoose == "scissors" && aiChoose == "stone") || // User wins Computer
+    (userChoose == "paper" && aiChoose == "scissors") ||
+    (userChoose == "stone" && aiChoose == "paper")
+  ) {
+    totalPlays = totalPlays + 1;
+    userScore = userScore + 1;
+    console.log(userScore);
+    myOutputValue =
+      "You have won! <br><br> You have won " +
+      userScore +
+      " out of " +
+      totalPlays +
+      " times ";
+  }
+  if (userChoose == "score check") {
+    myOutputValue =
+      " You have won " + userScore + " out of " + totalPlays + " times";
+  }
+
+  return myOutputValue;
+};
+
+// ///  Checking your score ///
+// var scoreCheck = function () {
+//   myOutputValue =
+//     " You have won " + userScore + " out of " + totalPlays + " times";
+// };
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  input = input.toLowerCase();
+  /// Creating a username ///
+  if (currentGameMode == "waiting for user name") {
+    userName = input;
+    console.log("Hi " + userName);
+    myOutputValue =
+      "Hi " +
+      userName +
+      " <br><br> Please Choose a mode. <br> <br>'Normal' or ' Reverse' ";
+  }
+  // Choosing Game Mode
+  if (input == "normal") {
+    currentGameMode = "normal";
+  } else if (input == "reverse") {
+    currentGameMode = "reverse";
+  }
+  console.log("Playing " + currentGameMode);
+  // Run the specific game the user wants
+
+  if (currentGameMode == "normal") {
+    myOutputValue = normalSPS(input);
+  }
+  if (currentGameMode == "reverse") {
+    myOutputValue = reverseSPS(input);
+  }
```

Comment:
perhaps you could combine the latter lines of code with the former ones? For example:

if (input == "normal") {
    currentGameMode = "normal";
myOutputValue = normalSPS(input);
  } else {
    currentGameMode = "reverse";
myOutputValue = reverseSPS(input);
  }